src/awkward/_v2/contents/regulararray.py

```diff
@@ -459,3 +459,87 @@ def _localindex(self, axis, depth):
                 self._identifier,
                 self._parameters,
             )
+
+    def _combinations(self, n, replacement, recordlookup, parameters, axis, depth):
+        if n < 1:
+            raise ValueError("in combinations, 'n' must be at least 1")
+        posaxis = self._axis_wrap_if_negative(axis)
+        if posaxis == depth:
+            return self._combinations_axis0(n, replacement, recordlookup, parameters)
+        elif posaxis == depth + 1:
+            if (
+                self.parameter("__array__") == '"string"'
+                or self.parameter("__array__") == '"bytestring"'
+            ):
+                raise ValueError(
+                    "ak.combinations does not compute combinations of the characters of a string; please split it into lists"
+                )
+
+            size = self._size
+            if replacement:
+                size = size + (n - 1)
+            thisn = n
+            if thisn > size:
+                combinationslen = 0
+            elif thisn == size:
+                combinationslen = 1
+            else:
+                if thisn * 2 > size:
+                    thisn = size - thisn
+                combinationslen = size
+                for j in range(2, thisn + 1):
+                    combinationslen = combinationslen * (size - j + 1)
+                    combinationslen = combinationslen // j
+
+            totallen = combinationslen * len(self)
+            nplike_tocarryraw = self.nplike.empty(n, dtype=np.intp)
+            tocarry = []
+            for i in range(n):
+                ptr = ak._v2.index.Index64.empty(totallen, self.nplike, dtype=np.int64)
+                tocarry.append(ptr)
+                nplike_tocarryraw[i] = ptr.ptr
+
+            tocarryraw = ak._v2.contents.numpyarray.NumpyArray(nplike_tocarryraw)
+            toindex = ak._v2.index.Index64.empty(n, self.nplike, dtype=np.int64)
+            fromindex = ak._v2.index.Index64.empty(n, self.nplike, dtype=np.int64)
+
+            if self._size != 0:
+                self._handle_error(
+                    self.nplike[
+                        "awkward_RegularArray_combinations_64",
+                        np.int64,
+                        toindex.to(self.nplike).dtype.type,
+                        fromindex.to(self.nplike).dtype.type,
+                    ](
+                        tocarryraw.data,
```

Comment:
Again with `nplike_tocarryraw` and `tocarryraw`.

Reply:
Yup, rectified the issue.